lib/mayaUsd/ufe/UsdUndoDuplicateCommand.cpp

```diff
@@ -59,6 +60,8 @@ void UsdUndoDuplicateCommand::primInfo(
     SdfPath&        usdDstPath,
     SdfLayerHandle& srcLayer)
 {
+    ufe::applyCommandRestriction(srcPrim, "duplicate");
```

Comment:
1- both static flavors of primInfo() can go away.  All the logics in primInfo can be moved directly to the constructor. 

https://github.com/Autodesk/maya-usd/blob/5ece36c5c45449b348d76ae73523e25179dc984f/lib/mayaUsd/ufe/UsdUndoDuplicateCommand.h#L61

https://github.com/Autodesk/maya-usd/blob/5ece36c5c45449b348d76ae73523e25179dc984f/lib/mayaUsd/ufe/UsdUndoDuplicateCommand.h#L53

2- fStage can go away. We don't need to save it . It can be obtained from fUfeSrcPath. 

3- I believe there is a bug around the layer. You should get layer from the srcPrim and store in 

`fLayer = GetStage()->GetEditTarget().GetLayer()`

Reply:
Nit-pick: these includes can go away.

#include <mayaUsd/ufe/UsdSceneItem.h>
#include <pxr/usd/usd/stage.h>

all the fxxx member variables should follow the coding standard.

Reply:
1. primInfo() doesn't even have an implementation as far as I can tell, so I'll just get rid of it.  The other primInfo static is used by Scene Item Ops so can't be moved to the constructor.

2. Yes, we can get the stage when needed.  I think I'd have to use _srcPrim.GetStage() though, fUfeSrcPath doesn't know anything about Usd.

3. Hmm.  fLayer is currently set by passing it to primInfo where it's set to the whatever layer the def primspec is on.  What you're suggesting is to change this to always grab the current edit target layer?  With the new restriction we want to enforce, I think those two should always be the same, but there is a comment in the next function down that says we chose to not just grab the edit target layer.

4. I removed one include that didn't look like it was needed.  I only see stage.h in the .cpp though, which is needed.

5. Yes, I can switch to the new naming convention for variables while I'm here.